### PR TITLE
layers: Handle PhysicalStorageBuffer to itself

### PR DESCRIPTION
**Pull request by sjfricke**
Should fix https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4255#issuecomment-1225956592 (cc @Tony-LunarG the CTS ran for me without crashing now too)

### CODE REVIEW

**Comment by lunarpapillo** [comment]
CI failure was likely due to a driver issue.  Re-running.

**Review by Tony-LunarG — APPROVED** [review]
LGTM - fixes the CTS problems I saw

File: tests/positive/pipeline.cpp [hunk 1]
```diff
@@ -4232,6 +4232,55 @@ void main() {
...
+
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported";
+    }
+
+    auto features12 = LvlInitStruct<VkPhysicalDeviceVulkan12Features>();
+    auto features2 = GetPhysicalDeviceFeatures2(features12);
+    if (VK_TRUE != features12.bufferDeviceAddress) {
+        printf("%s VkPhysicalDeviceVulkan12Features::bufferDeviceAddress not supported and is required. Skipping.\n", kSkipPrefix);
+        return;
```
Comment (ncesario-lunarg) [inline tests/positive/pipeline.cpp:4254]: `GTEST_SKIP`?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**